FaliJedan/FaliJedan.Web/ClientApp/src/components/EventCard.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from "react";
+import LocationDisplay from "./Map/LocationDisplay";
+import EventCardInfo from "./EventCardInfo";
+import { Swipeable } from "react-swipeable";
+import { LEFT, RIGHT, HOST_NAME_CLASS } from "../constants";
+
+class EventCard extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isMapActive: false,
+      isDetailsActive: false
+    };
+  }
+
+  swipe = direction => {
+    if (
+      (direction === LEFT && !this.state.isMapActive) ||
+      (direction === RIGHT && this.state.isMapActive)
+    ) {
+      this.toggleMap();
+    }
+  };
+
+  toggleMap = () => {
+    this.setState(prevState => {
+      const isMapActive = !{ ...prevState }.isMapActive;
+      return { isMapActive };
+    });
+  };
+
+  toggleDetails = event => {
+    if (event.target.className === HOST_NAME_CLASS) {
```

Comment:
Why not handle presses on the name with handleNameClick(e) and e.stopPropagation()?